src/traces/ohlc/helpers.js

```diff
@@ -95,14 +95,65 @@ exports.makeTransform = function(traceIn, state, direction) {
 };
 
 exports.getFilterFn = function(direction) {
+    return new _getFilterFn(direction);
+};
+
+function _getFilterFn(direction) {
+    var isPrevThisDirection = null;
+    var cPrev = null;
+    var fn;
+
     switch(direction) {
         case 'increasing':
-            return function(o, c) { return o <= c; };
+            fn = function(o, c) {
+                if(o === c) {
```

Comment:
We haven't cast to numbers, have we? Looks like this will break if we mix strings and numbers (or use different string representations of the same number, like `'2.0'` and `'2.00'`

I guess a point should get dropped if *any* of o, h, l, c is not numeric?